### PR TITLE
[macOS .NET] Add homebrew paths to the search locations.

### PR DESCRIPTION
**Pull request by bruvzg**
See https://github.com/godotengine/godot/issues/73747

### CODE REVIEW

File: modules/mono/editor/hostfxr_resolver.cpp [hunk 1]
```diff
@@ -198,7 +198,13 @@ bool get_default_installation_dir(String &r_dotnet_root) {
...
 	return true;
 #elif defined(MACOS_ENABLED)
-	r_dotnet_root = "/usr/local/share/dotnet";
+	if (DirAccess::exists("/opt/homebrew/opt/dotnet/libexec/host/fxr")) {
+		r_dotnet_root = "/opt/homebrew/opt/dotnet/libexec";
+	} else if (DirAccess::exists("/usr/local/opt/dotnet/libexec/host/fxr")) {
+		r_dotnet_root = "/usr/local/opt/dotnet/libexec";
+	} else if (DirAccess::exists("/usr/local/share/dotnet/host/fxr")) {
+		r_dotnet_root = "/usr/local/share/dotnet";
+	}
```
Comment (raulsntos) [inline modules/mono/editor/hostfxr_resolver.cpp:207]: AFAIK, the `hostfxr_resolver` is supposed to mirror the [`fxr_resolver`](https://github.com/dotnet/runtime/blob/6e422d6f6eb6bac07ea066d06fe42856459a5d79/src/native/corehost/fxr_resolver.cpp) from .NET so I wouldn't add these here.

The .NET implementation seems to only return `/usr/local/share/dotnet`:

https://github.com/dotnet/runtime/blob/6e422d6f6eb6bac07ea066d06fe42856459a5d79/src/native/corehost/hostmisc/pal.unix.cpp#L581-L594

File: modules/mono/editor/GodotTools/GodotTools/Build/DotNetFinder.cs [hunk 2]
```diff
@@ -32,6 +32,22 @@ public static string FindDotNetExe()
...
+                    }
+                }
+                else
+                {
+                    string dotnet_hb = "/opt/homebrew/opt/dotnet/libexec"; // Look for ARM homebrew version.
+                    if (File.Exists(dotnet_hb))
+                    {
+                        return dotnet_hb;
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (bruvzg) [reply modules/mono/editor/GodotTools/GodotTools/Build/DotNetFinder.cs:50]: It is. `if (OS.IsMacOS)` is on line 20.

[the remark this answers is not in the document]
Comment (raulsntos) [reply modules/mono/editor/GodotTools/GodotTools/Build/DotNetFinder.cs:50]: Oh sorry, I totally missed it.

File: modules/mono/editor/hostfxr_resolver.cpp (hunk 1 above)
Comment (RedworkDE) [reply modules/mono/editor/hostfxr_resolver.cpp:207]: Without this, this won't actually help anything, as it still won't be using the homebrew installed dotnet as the editor runtime.

Comment (raulsntos) [reply modules/mono/editor/hostfxr_resolver.cpp:207]: I was just wondering if it should be added somewhere else. But IIRC we try to find hostfxr from the `dotnet` found in PATH, is `dotnet` not in the PATH when installed using homebrew?

Comment (bruvzg) [reply modules/mono/editor/hostfxr_resolver.cpp:207]: macOS apps ignore all user defined environment variables, `PATH` is always `/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin` (only `/usr/local/bin` is writable).

**Comment by zengxs** [comment]
I think there is another scenario that should not be overlooked.

If .NET is installed using the [`dotnet-install.sh`](https://dotnet.microsoft.com/en-us/download/dotnet/scripts) script provided by Microsoft, it will be installed by default in the `$HOME/.dotnet` directory.

Therefore, I suggest that we should also search this directory additionally.

File: modules/mono/editor/GodotTools/GodotTools/Build/DotNetFinder.cs [hunk 3]
```diff
@@ -32,6 +32,22 @@ public static string FindDotNetExe()
...
+                {
+                    string dotnet_hb = "/usr/local/opt/dotnet/libexec"; // Look for Intel homebrew version.
+                    if (File.Exists(dotnet_hb))
+                    {
+                        return dotnet_hb;
+                    }
+                }
+                else
+                {
+                    string dotnet_hb = "/opt/homebrew/opt/dotnet/libexec"; // Look for ARM homebrew version.
```
[the remark this answers is not in the document]
Comment (bruvzg) [reply modules/mono/editor/GodotTools/GodotTools/Build/DotNetFinder.cs]: `/opt/homebrew/opt/dotnet` is always a symlink to `/opt/homebrew/Cellar/dotnet/{current_version}` so there's no reason to check for specific versions.

**Comment by a-johnston** [comment]
I found this topic after installing dotnet 9 through homebrew, forgetting that I had used the microsoft installer previously. I think this would be still valuable and not really obsoleted by either loading an env from zsh or by having an editor option for where to look for dotnet, considering that these are standard paths and that not everyone uses zsh. I rebased + built this locally and it worked as expected.

**Comment by bruvzg** [comment]
Closing in favor of #96146 + #81266 (this combination should work with any type of installation).

**Pull request closed without merging**